media/CircleEditor/view-sidebar.js

```diff
@@ -0,0 +1,2508 @@
+                this._addInput(input.name, input);
+            }
+        }
+
+        const outputs = node.outputs;
+        if (outputs && outputs.length > 0) {
+            this._addHeader('Outputs');
+            for (const output of outputs) {
+                this._addOutput(output.name, output);
+            }
+        }
+
+        const separator = this._host.document.createElement('div');
+        separator.className = 'sidebar-view-separator';
+        this._elements.push(separator);
+    }
+
+    render() {
+        return this._elements;
+    }
+
+    _addHeader(title) {
+        const headerElement = this._host.document.createElement('div');
+        headerElement.className = 'sidebar-view-header';
+        headerElement.innerText = title;
+        this._elements.push(headerElement);
+    }
+
+    _addProperty(name, value) {
+        const item = new sidebar.NameValueView(this._host, name, value);
+        this._elements.push(item.render());
+    }
+
+    _addAttribute(name, attribute) {
+        const item = new NodeAttributeView(this._host, attribute);
+        item.on('show-graph', (sender, graph) => {
+            this._raise('show-graph', graph);
+        });
+        const view = new sidebar.NameValueView(this._host, name, item);
+        this._attributes.push(view);
+        this._elements.push(view.render());
+    }
+
+    _addInput(name, input) {
+        if (input.arguments.length > 0) {
+            const view = new sidebar.ParameterView(this._host, input);
+            view.on('export-tensor', (sender, tensor) => {
+                this._raise('export-tensor', tensor);
+            });
+            view.on('error', (sender, tensor) => {
+                this._raise('error', tensor);
+            });
+            const item = new sidebar.NameValueView(this._host, name, view);
+            this._inputs.push(item);
+            this._elements.push(item.render());
+        }
+    }
+
+    _addOutput(name, output) {
+        if (output.arguments.length > 0) {
+            const item = new sidebar.NameValueView(
+                this._host, name, new sidebar.ParameterView(this._host, output));
+            this._outputs.push(item);
+            this._elements.push(item.render());
+        }
+    }
+
+    toggleInput(name) {
+        for (const input of this._inputs) {
+            if (name === input.name) {
+                input.toggle();
+            }
+        }
+    }
+
+    on(event, callback) {
+        this._events = this._events || {};
+        this._events[event] = this._events[event] || [];
+        this._events[event].push(callback);
+    }
+
+    _raise(event, data) {
+        if (this._events && this._events[event]) {
+            for (const callback of this._events[event]) {
+                callback(this, data);
+            }
+        }
+    }
+};
+
+/**
+ * sidebar.EditAttributesView Class
+ */
+
+/**
+ * sidebar.EditInputsView Class
+ */
+
+/**
+ * sidebar.EditOutputsView Class
+ */
```

Comment:
These comments are not that good because other members are hard to know what is this :(
Please write like following example..!

```suggestion
/**
 * TODO Implement sidebar.EditAttributesView Class
 */

/**
 * TODO Implement sidebar.EditInputsView Class
 */

/**
 * TODO Implement sidebar.EditOutputsView Class
 */
```